File: cmd/feature-experimentation/analyze/code-samples/openfeature/go/sample-OF.go
package main

import (
	"context"
	"fmt"

	"github.com/open-feature/go-sdk/openfeature"
)

func main() {
	// Register your feature flag provider
	openfeature.SetProviderAndWait(openfeature.NoopProvider{})
	// Create a new client
	client := openfeature.NewClient("app")
	// Evaluate your feature flag
	v2Enabled := client.Boolean(
		context.Background(), "of_go_v2_enabled", true, openfeature.EvaluationContext{},
	)

	// Use the returned flag value
	if v2Enabled {
		fmt.Println("v2 is enabled")
	}
}

File: cmd/feature-experimentation/analyze/code-samples/openfeature/js/sample-OF.ts
import { OpenFeature } from "@openfeature/server-sdk";

// Register your feature flag provider
await OpenFeature.setProviderAndWait(new YourProviderOfChoice());

// create a new client
const client = OpenFeature.getClient();

// Evaluate your feature flag
const v2Enabled = await client.getBooleanValue("of_ts_v2_enabled", false);

if (v2Enabled) {
  console.log("v2 is enabled");
}
